boards/arm/nucleo_f411re/pinmux.c

```diff
@@ -40,6 +40,14 @@ static const struct pin_config pinconf[] = {
 	{STM32_PIN_PA6, STM32F4_PINMUX_FUNC_PA6_SPI1_MISO},
 	{STM32_PIN_PA7, STM32F4_PINMUX_FUNC_PA7_SPI1_MOSI},
 #endif	/* CONFIG_SPI_1 */
+#ifdef CONFIG_I2S_1
+	{STM32_PIN_PA4, STM32F4_PINMUX_FUNC_PA4_I2S1_WS},
+	{STM32_PIN_PA5, STM32F4_PINMUX_FUNC_PA5_I2S1_CK},
+	{STM32_PIN_PA7, STM32F4_PINMUX_FUNC_PA7_I2S1_SD},
+#endif	/* CONFIG_I2S_1 */
+	/* TBD ???? */
+	{STM32_PIN_PB4, (STM32_PINMUX_ALT_FUNC_2 | STM32_PUSHPULL_NOPULL | STM32_OSPEEDR_VERY_HIGH_SPEED)},
```

Comment:
What are you trying to do here?

Reply:
@erwango 
You touched the point. As I highlighted in the notes this changes to the board are not
proper and must be fixed. But I prepared the PR anyway so that it is evident what is my
intention.

Basically, to make the x_nucleo_cca02m1 shield working fine (and in both mono and 
stereo mode) there are few things that must be configured, but the only way I found
was in the board itself, which is not proper.

The shield needs:
- I2S configuration (PLL and so on)
- DMA configuration
- timer configuration for proper microphone clocking in stereo mode (timer generates clock which
goes out on pins)

In this file the PB4 and PB5 are the pins where a 2x clock is generated using timer as explained
in AN5027.

I guess this part should be either in board/sheilds or in sample/shields. 
Or maybe here under ifdef. Any suggestion?

Reply:
>  Any suggestion?

What about using gpio map and do this configuration inside the sample code?
define in dts:
```
gpio-d2 = <&arduino_header 7 0>;
```
Then, in sample code, use gpio-d2 generated define to do the gpio configuration?
This is basically what is done for cs-gpio for SPI controllers.

Reply:
@avisconti , did you checked above proposal?

Reply:
@erwango, no, not yet.
I did try to understand the steps to do that looking at the example, but stopped after few
doubts. I'll be back on this later on.